Plugins/Public/balancemagic/Main.cpp

```diff
@@ -198,62 +227,68 @@ bool UserCmd_Process(uint iClientID, const wstring &wscCmd)
 	return false;
 }
 
-void __stdcall HkCb_AddDmgEntry(DamageList *dmg, ushort subObjID, float& setHealth, DamageEntry::SubObjFate fate)
+void __stdcall HkCb_AddDmgEntry(DamageList *dmg, ushort subObjID, float& setHealth, DamageEntry::SubObjFate& fate)
 {
 	returncode = DEFAULT_RETURNCODE;
-	if (iDmgToSpaceID && iDmgMunitionID)
+	if (!iDmgMunitionID || (!iDmgToSpaceID && !iDmgTo))
 	{
-		map<uint, DamageMultiplier>::iterator iter = mapDamageAdjust.find(iDmgMunitionID);
-		if (iter != mapDamageAdjust.end())
-		{
-			float curr, max;
-			bool bShieldsUp;
-
-			if (subObjID == 1) // 1 is base (hull)
-				pub::SpaceObj::GetHealth(iDmgToSpaceID, curr, max);
-			else if (subObjID == 65521) // 65521 is shield (bubble, not equipment)
-				pub::SpaceObj::GetShieldHealth(iDmgToSpaceID, curr, max, bShieldsUp);
-			else
-				return; // If hit mounted equipment - do not continue with uninitialized variables.
+		return;
+	}
+	const auto& iter = mapDamageAdjust.find(iDmgMunitionID);
+	if (iter == mapDamageAdjust.end())
+	{
+		return;
+	}
 
-			uint iTargetType;
-			pub::SpaceObj::GetType(iDmgToSpaceID, iTargetType);
+	const DamageMultiplier& dmgInfo = iter->second;
+	float curr, max;
+	bool bShieldsUp;
+
+	if (subObjID == 1) // 1 is base (hull)
+		pub::SpaceObj::GetHealth(iDmgToSpaceID, curr, max);
+	else if (subObjID == 65521) // 65521 is shield (bubble, not equipment)
+		pub::SpaceObj::GetShieldHealth(iDmgToSpaceID, curr, max, bShieldsUp);
+	else if (subObjID <= 32) // collision groups
```

Comment:
don't suppose we have a relevant constant to compare to regarding max collision group IDs? I dislike these magic numbers

Reply:
hullID being 1 is handled above, and engine/powercore have no hitboxes, so a simple < 34 handles all possible collision groups already